### PR TITLE
[Relay] QLinearMatMul allows 1D weight_scale, weight_zero_point inputs

### LINKED ISSUE #9908: [Bug] [ONNX] QLinearMatMul works with binary output only
I'm using [`from_onnx`](https://github.com/apache/tvm/blob/44fe7ef816565f43380c50e0b43fd626fad9d029/python/tvm/relay/frontend/onnx.py#L5063) to convert my model 
[dense_h32_w32_c3_sNone_pNone_kNone.pt_quant.onnx.zip](https://github.com/apache/tvm/files/7851265/dense_h32_w32_c3_sNone_pNone_kNone.pt_quant.onnx.zip) into tvm relay. It looks like `b_scale` in [`QLinearMatMul`](https://github.com/apache/tvm/blob/44fe7ef816565f43380c50e0b43fd626fad9d029/python/tvm/relay/frontend/onnx.py#L3760) expects a scalar scale not a vector of scales. 
I'm not sure if this is a problem with onnx creating a vector for the weights, or if QLinearMatMul should support scale vectors.

Info about the model:
1. Input: (batch, channel, height, width) -> (1, 3, 32, 32)
2. Global Average pool: (batch, channel, 1, 1) -> (1, 3, 1, 1)
3. Reshape: (batch, channel) -> (1, 3)
4. Dense Layer: (batch, channel, 20) -> (3, 20) 
    > Dense Layer is where the error occurs 

> You might notice when you debug `b_scale`'s shape that it has shape=20. 

### Expected behavior

No errors converting my model [dense_h32_w32_c3_sNone_pNone_kNone.pt_quant.onnx.zip](https://github.com/apache/tvm/files/7851265/dense_h32_w32_c3_sNone_pNone_kNone.pt_quant.onnx.zip) into tvm relay using  [`from_onnx`](https://github.com/apache/tvm/blob/44fe7ef816565f43380c50e0b43fd626fad9d029/python/tvm/relay/frontend/onnx.py#L5063).

### Actual behavior

```
assert num_elem == 1, "Cannot squeeze tensor shape {} to scalar form.".format(x_shape)
E       AssertionError: Cannot squeeze tensor shape (20,) to scalar form.
```

### PR DESCRIPTION
**Pull request by FZYUAN-1**
Hello,

This PR is to fix #9908. Previously I made a mistake in git (https://github.com/apache/tvm/pull/9946), now this is a clean PR.

@AndrewZhaoLuo Thank you for your code review, very valuable advice !

### CODE REVIEW

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -3804,10 +3804,13 @@ def _impl_v10(cls, inputs, attr, params):
         #
         # This function attempts to present 'x' in a form that meets both of those
         # requirements.
-        def try_resolve_to_const_scalar(x, dtype_override=None):
+        def try_resolve_to_const(x, dtype_override=None):
             x2 = try_resolve_var_to_const(x, params)
-            x3 = ensure_scalar_shape(x2)
-
+            num_elem = np.prod(infer_shape(x))
+            if num_elem == 1:
```
Comment (AndrewZhaoLuo) [inline python/tvm/relay/frontend/onnx.py:3810]: nit: This can be 
```
if num_elem == 1:
    x2 = ensure_scalar_shape(x2)
```
And we dont need x3 now

**Review by AndrewZhaoLuo — APPROVED** [review]
LGTM, just a small nit

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (FZYUAN-1) [reply python/tvm/relay/frontend/onnx.py:3810]: Thanks, done

**Review by AndrewZhaoLuo — COMMENTED** [review]
@FZYUAN-1 

Looks like CI bugged out (we recently moved servers I think). You can redo CI by pushing an empty commit e.g. `git commit -m 'jostle' --allow-empty` and `git push`

After CI passes I will merge

**Comment by FZYUAN-1** [comment]
> @FZYUAN-1
> 
> Looks like CI bugged out (we recently moved servers I think). You can redo CI by pushing an empty commit e.g. `git commit -m 'jostle' --allow-empty` and `git push`
> 
> After CI passes I will merge

Thanks. Done.

**Comment by AndrewZhaoLuo** [comment]
Need to jostle again, looks like the CI folks did not enable autoscaling until now so your job timed out.

**Comment by FZYUAN-1** [comment]
> Need to jostle again, looks like the CI folks did not enable autoscaling until now so your job timed out.

Done.

**Pull request merged**